raft-tech/TANF-app
# Skip clamav locally

## Summary of Changes
_Provide a brief summary of changes_
To improve spin-up time, I have set an environment flag for the Django side of our data file scans (CLAMAV_NEEDED). As a TODO, I need to potentially address the conditional spin-up of clamav in docker compose.

## How to Test
_List the steps to test the PR_

1. Setup environment variables, load aliases, and clean docker caches before new build/start.
```bash
export TDRS_HOME=`pwd`
source commands.sh
tdrs-prune-all-docker-data
echo "CLAMAV_NEEDED=False" >> tdrs-backend/.env
export CLAMAV_NEEDED="False"
tdrs-start
```
1. Using `docker ps`, ensure neither ZAP nor CLAMAV images were created or spun up

1. Open http://localhost:3000/ and sign in.
1. Ensure there are no redirect loops.
1. Ensure 
> *Demo GIF(s) and screenshots for testing procedure*
![image](https://user-images.githubusercontent.com/84722778/157717678-59448c98-76b6-45d9-bcff-4b8646825b77.png)

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Prevent unnecessary waiting and spinup of CLAMAV and ZAP unless developer needs.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_93.39%_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_97%_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

_n/a_

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

_n/a_

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

_n/a_

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

_n/a_

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

_n/a_
